[TECH] Ajout stylistic dans le front et les scripts (PIX-20125)

mcampourcy opened the pull request:
## 🍂 Problème

Avec le passage d’ESLint en V9, beaucoup de règles sont dépréciées et seront supprimées en V11. Plus d’infos [par ici](https://eslint.org/blog/2023/10/deprecating-formatting-rules/).

## 🌰 Proposition

Ajouter et configurer le plugin stylistic pour gérer ces règles dans le front et les scripts

## 🍁 Remarques

- Pour avoir un aperçu des règles dépréciées, il est possible de lancer la commande npx eslint --inspect-config
- Le plugin stylistic reprend les règles de lint qu'on utilisait avant (et qui sont dépréciées) + quelques règles prettier qui étaient utiles

## 🪵 Pour tester

CI au vert

pix-bot-github commented:
Choisir les applications à déployer :

- [x] [pix-lcms](https://pix-lcms-review-pr1200.osc-fr1.scalingo.io/) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-lcms-review-pr1200/environment) <!-- pix-lcms-review -->

mcampourcy committed:
tech(scripts-dependencies): Add stylistic in scripts to manage deprecated rules from eslint

Eithliu commented on pix-editor/app/components/alternatives.hbs:
```diff
@@ -1,9 +1,12 @@
-<div class="main-title  {{if this.config.lite "lite" ""}}">
+<div class="main-title {{if this.config.lite 'lite' ''}}">
```
Pas sûre que la règle sur l'override dans les fichiers hbs fonctionne, du coup ^^'

mcampourcy replied:
Je pense que c'est lié aux quotes dans les quotes, et qu'on a des simple quote pour ne pas entrer en conflit avec les double 🤔

Eithliu commented on scripts/compare-challenges-and-attachments/index.js:
```diff
@@ -14,21 +14,25 @@ async function main() {
-        console.error(`Found inconsistent challenge and attachments: challenge ${challenge.id} should have these attachments ${JSON.stringify(errors)}`);
+        console.error(
+          `Found inconsistent challenge and attachments: challenge ${challenge.id} should have these attachments ${JSON.stringify(errors)}`,
+        );
```
Une ligne sur 3 lignes pour un bête console.error 😢

mcampourcy replied:
Dans ce cas précis, la ligne fait 174 caractères. Sur mon IDE, ça donne ça, mais c'est lié à mon affichage : 

```js
console.error(`Found inconsistent challenge and attachments: challenge ${challenge.id} should have these attachments ${JSON.stringify(errors)
}`);
```

(après, j'aime bien l'affichage sur 3 lignes du console.error, mais c'est un avis perso :))

Pour éviter ce genre de saut à la ligne, on peut monter le nb de caractères max par ligne à 180. Je le note, et je vois si c'est suffisant en fonction des autres remontées

Jeremiejade commented on pix-editor/app/components/competence/competence-footer.js:
```diff
@@ -13,18 +12,14 @@ export default class CompetenceFooter extends Component {
-    return this.displayProductionStats
-      ? this.args.competence.productionTubeCount
-      : this.args.competence.tubeCount;
+    return this.displayProductionStats ? this.args.competence.productionTubeCount : this.args.competence.tubeCount;
```
pouvoir mettre les ternaires avec des retours à la ligne

mcampourcy replied:
Prettier va inliner les ternaires tant qu'elles sont en-dessous de la taille max d'une ligne (actuellement: 120 caractères)

Je regarde si on peut faire sauter cette obligation, c'est pas garanti

mcampourcy commented on pix-editor/app/components/competence/prototypes/challenge-header.hbs:
```diff
@@ -1,16 +1,20 @@
       {{#if @maximized}}
-        <button class="ui icon button item" {{on "click" @minimize}} type="button"><i class="window minimize icon"></i></button>
+        <button class="ui icon button item" {{on "click" @minimize}} type="button"><i
+            class="window minimize icon"
```
🔥

Jeremiejade commented on pix-editor/app/components/alternatives.hbs:
```diff
@@ -1,9 +1,12 @@
-      <LinkTo @route="authenticated.competence.prototypes.single" @model={{@challenge}} class="ui button icon item"><i class="icon window close"></i></LinkTo>
+      <LinkTo @route="authenticated.competence.prototypes.single" @model={{@challenge}} class="ui button icon item"><i
+          class="icon window close"
+        ></i></LinkTo>
```
```suggestion
 <LinkTo @route="authenticated.competence.prototypes.single" @model={{@challenge}} class="ui button icon item">
   <i class="icon window close" ></i>
  </LinkTo>
```
ce format me semble plus logique, peut être qu'il est perdu parce qu'on a pas mis de retours à la ligne ?

mcampourcy replied:
Je pense que oui, ça a planté pour ça. Je vais modifier à la main

Jeremiejade commented on pix-editor/app/components/competence/competence-footer.hbs:
```diff
@@ -22,7 +32,8 @@
     <button class="ui button right item" {{on "click" @newPrototype}} type="button">
-      <i class="plus square outline icon" data-test-create-new-challenge></i> Nouveau prototype
+      <i class="plus square outline icon" data-test-create-new-challenge></i>
+      Nouveau prototype
```
ça ne risque pas de casser l'affichage de faire ça ?

Jeremiejade replied:
ça semble ok

mcampourcy replied:
yes, j'ai regardé aussi, ça a l'air OK

Jeremiejade commented on pix-editor/app/components/competence/grid/cell-production.hbs:
```diff
@@ -1,10 +1,5 @@
-    @model={{@skillOverview.prototypeId}}
-    class="skill-cell__link"
-  >
+  <LinkTo data-test-skill-cell-link @route={{@link}} @model={{@skillOverview.prototypeId}} class="skill-cell__link">
```
pk ça a été oneliné ça ?

mcampourcy replied:
ça prend moins de 120 caractères (taille max d'une ligne)

Eithliu commented on scripts/compare-releases-integrity/index.test.js:
```diff
@@ -177,123 +165,120 @@ describe('Scripts | Compare Release Integrity', function() {
+            },
+            {
+              id: '2',
+            },
+            {
+              id: '3',
+            },
+          ],
+        },
```
❤️ ❤️ ❤️ ❤️

Eithliu commented on scripts/copy-skills-and-set-challenges-as-focusable/airtable-fields.js:
```diff
@@ -3,30 +3,31 @@ export const USEFUL_SKILL_FIELDS = [
   'Indice fr-fr',
   'Indice en-us',
-  'Statut de l\'indice',
+  "Statut de l'indice",
```
J'ai pas de souci avec ça, mais on a toujours échappé les `\'` du coup, je sais pas si on veut garder notre fonctionnement ou accepter que ça transforme les single quotes en double quotes quand on a des apostrophes.

Jeremiejade commented on pix-editor/app/components/form/static-course.js:
```diff
@@ -153,36 +157,38 @@ class NameField extends FormField {
-    return challengeIds
-      .map((challengeId) => challengeId.trim())
-      .filter((challengeId) => challengeId.length > 0);
+    return challengeIds.map((challengeId) => challengeId.trim()).filter((challengeId) => challengeId.length > 0);
```
ici on voudrait garder le format d'avant

Jeremiejade commented on pix-editor/app/components/statistics/production.js:
```diff
@@ -19,9 +18,12 @@ export default class StatisticsProductionComponent extends Component {
+          tube.liveSkills.reduce((current, skill) => {
+            return current + skill.challengesArray.filter((challenge) => challenge.isValidated).length;
+          }, 0)
+        );
```
bon le code de base est un peut foufou, mais la proposition de prettier n'est pas claire

Jeremiejade commented on pix-editor/app/components/statistics/production.js:
```diff
@@ -41,7 +43,7 @@ export default class StatisticsProductionComponent extends Component {
       skills: this.productionSkillCounts[code],
       challenges: this.productionChallengeCounts[code],
-      rate: (this.productionChallengeCounts[code] * 100 / this.productionChallengeTotal).toFixed(1),
+      rate: ((this.productionChallengeCounts[code] * 100) / this.productionChallengeTotal).toFixed(1),
```
il n'y a pas besoin de l'ajout de ces parenthèse :/

mcampourcy commented on pix-editor/app/components/form/mission.hbs:
```diff
@@ -93,44 +92,37 @@
 
-      <label class="new-mission-form__label-text-area" for="mission-introduction-media-alt">Texte alternatif pour le média
-        d'introduction</label>
+      <label class="new-mission-form__label-text-area" for="mission-introduction-media-alt">Texte alternatif pour le
```
checker ce saut de ligne

mcampourcy commented on pix-editor/app/components/form/mission.hbs:
```diff
@@ -93,44 +92,37 @@
-        {{on "change" this.updateDocumentationUrl}}
-      />
+    <label class="new-mission-form__label-text-area" for="mission-documentation-url">URL de la documentation de la
+      mission</label>
```
:fire: Saut de ligne

Eithliu commented on scripts/enable-shuffled-on-challenges/index.js:
```diff
@@ -120,10 +124,7 @@ function* chunks(array, chunkSize) {
-    ['ID Épreuve'],
-    ...challenges.map((challenge) => [challenge.id]),
-  ]);
+  const ws = xlsxUtils.aoa_to_sheet([['ID Épreuve'], ...challenges.map((challenge) => [challenge.id])]);
```
En général, quand on fait un spread operator, on met à la ligne pour une meilleure lisibilité. Du coup, j'pense que ce serait plus cool si on pouvait choisir nous même de mettre à la ligne.

Eithliu commented on scripts/enable-shuffled-on-challenges/index.js:
```diff
@@ -137,10 +138,7 @@ function createAirtableClient({ apiKey, base }) {
-    AIRTABLE_API_KEY: airtableApiKey,
-    AIRTABLE_BASE: airtableBase,
-  } = process.env;
+  const { AIRTABLE_API_KEY: airtableApiKey, AIRTABLE_BASE: airtableBase } = process.env;
```
Alors, ici, il one-line...

Eithliu commented on scripts/list-duplicated-attachments/index.js:
```diff
@@ -12,12 +12,16 @@ export async function listDuplicatedAttachments({ lcmsApiKey }) {
+  const {
+    content: { challenges },
+  } = await res.json();
+
```
...Mais ici, il met tout à la ligne 🤔

mcampourcy commented on pix-editor/app/components/pop-in/confirm-log.hbs:
```diff
@@ -25,13 +21,16 @@
     </PixButton>
-    <PixButton data-test-confirm-log-approve @triggerAction={{fn @onApprove this.changeLogValue}}>{{t 'common.validate'}}</PixButton>
+    <PixButton data-test-confirm-log-approve @triggerAction={{fn @onApprove this.changeLogValue}}>{{t
+        "common.validate"
```
🔥 saut de ligne

mcampourcy committed:
tech(scripts-dependencies): Apply stylistic formatting for scripts files

mcampourcy committed:
tech(pixeditor-dependencies): Add stylistic for pixeditor to manage deprecated rules from eslint

mcampourcy committed:
tech(pixeditor-dependencies): Apply stylistic formatting for pixeditor js files

HEYGUL committed:
chore: add prettier config file for front

prettier is built-in in some IDE, thus it makes sense to have a basic configuration file for it

HEYGUL commented:
J'ai ajouté un commit qui ajoute une configuration très basique de prettier afin de ne pas avoir de soucis avec les IDE qui formatte avec un prettier "built-in" (c'est notamment le cas de Zed)
